stable/application-mesh/templates/gateway.yaml

```diff
@@ -0,0 +1,48 @@
+{{- if .Values.global.mesh.enabled }}
+{{- range $_, $ingress := .Values.ingress }}
+{{- if ne $ingress.host "example.matic.com" }}
+{{- if or (eq $ingress.gateway "internal") (eq $ingress.gateway "external") }}
+{{- $host := required "Ingress host must be specified" $ingress.host  }}
+{{- $hosts := append ($ingress.additional_hosts | default $.Values.defaults.ingress.additional_hosts) $ingress.host }}
+---
+apiVersion: networking.istio.io/v1beta1
+kind: Gateway
+metadata:
+  name: {{ $host | replace "." "-" }}
+  namespace: istio-ingress
+  annotations:
+    # Do not create dns records for this resource
+    external-dns.alpha.kubernetes.io/controller: "ignored-resource"
```

Comment:
AFAIK - external dns using virtual service to populate dns records. It is worthwhile to check and investigate if a gateway would be a better option.

This particular annotation is configured on wildcard gateways, so we don't create `*.matic.com` dns records